### ISSUE TITLE
Polygon clipping performance

### ISSUE THREAD

**Issue by asinghvi17** [JuliaGeo/GeometryOps.jl#129]
For a lot of geospatial usecases, polygons of size > 100 are de rigueur.  GeometryOps' performance on these polygons is horrible, more than 10x slower than LibGEOS. 

This could probably be improved by a monotone chain optimization, which we should add.  This will probably also need the prepared geometry interface.

**Comment by skygering** [comment]
Polygons with holes are also pretty bad. For example:
```julia
import GeoInterface as GI
import LibGEOS as GO
p25 = GI.Polygon([[(0.0, 0.0), (5.0, 0.0), (5.0, 8.0), (0.0, 8.0), (0.0, 0.0)], [(4.0, 0.5), (4.5, 0.5), (4.5, 3.5), (4.0, 3.5), (4.0, 0.5)], [(2.0, 4.0), (4.0, 4.0), (4.0, 6.0), (2.0, 6.0), (2.0, 4.0)]])
p26 = GI.Polygon([[(3.0, 1.0), (8.0, 1.0), (8.0, 7.0), (3.0, 7.0), (3.0, 5.0), (6.0, 5.0), (6.0, 3.0), (3.0, 3.0), (3.0, 1.0)], [(3.5, 5.5), (6.0, 5.5), (6.0, 6.5), (3.5, 6.5), (3.5, 5.5)],  [(5.5, 1.5), (5.5, 2.5), (3.5, 2.5), (3.5, 1.5), (5.5, 1.5)]])
```
<img width="956" alt="Screenshot 2024-04-24 at 2 11 41 PM" src="https://github.com/JuliaGeo/GeometryOps.jl/assets/60117338/e20244dd-1f23-40e6-b9b1-e26c88a051f3">
<img width="875" alt="Screenshot 2024-04-24 at 2 18 14 PM" src="https://github.com/JuliaGeo/GeometryOps.jl/assets/60117338/a9dd098d-f918-4abb-b63d-3c53cde0956c">

**Comment by asinghvi17** [comment]
Huh!  We should add the test suite polygons as benchmarks as well, so that all the bases are covered.  That could be a good start to a PkgBenchmark type of thing, since the current benchmarks are more focused on scaling across number of vertices which there isn't really a good solution for (yet) in the benchmark world.

**Comment by skygering** [comment]
This is with ExactPredicates since I was just testing a few of things in the REPL.

**Comment by asinghvi17** [comment]
I just ran an example with the number of vertices increased (using `segmentize`) and found the following:
![download-10](https://github.com/JuliaGeo/GeometryOps.jl/assets/32143268/a73dc208-1ad2-4079-9ed5-5152b2cf80e8)
(this is pre-ExactPredicates but I will check that branch and post a plot as well)

**Comment by asinghvi17** [comment]
Huh...exactpredicates is really hitting performance hard here.
![download-11](https://github.com/JuliaGeo/GeometryOps.jl/assets/32143268/6a500349-595e-438f-bac6-23ba767d27fe)

**Comment by rafaqz** [comment]
The problem is scaling... not ExactPredicates

**Comment by asinghvi17** [comment]
Yeah, looks like it...on the bright side, we at least have a head start :D

**Comment by asinghvi17** [comment]
The code for that plot is in https://github.com/Caltech-OCTO/GeometryOps.jl/pull/8
